packages/headless/src/app/engine.ts

```diff
@@ -157,6 +158,9 @@ export function buildEngine<
 
   if (analytics) {
     const {analyticsClientMiddleware, ...rest} = analytics;
+    if (analytics.enabled && coveoAnalytics.donottrack.doNotTrack()) {
```

Comment:
If you're asking about the function itself, it should, here's its module:
```ts
import {hasNavigator} from './detector';

const doNotTrackValues = ['1', 1, 'yes', true];

export function doNotTrack(): boolean {
    return (
        hasNavigator() &&
        [
            (<any>navigator).globalPrivacyControl,
            (<any>navigator).doNotTrack,
            (<any>navigator).msDoNotTrack,
            (<any>window).doNotTrack,
        ].some((value) => doNotTrackValues.indexOf(value) !== -1)
    );
}

export default doNotTrack;
```

Reply:
What if instead of adding this `if` check here, we modified the initial configuration done at the beginning of this function to force the options `analytics.enabled = false` on the engine when the do not track header is set ?

With an info message in the console that explains that analytics were disabled because of the header.